client/src/components/Schedule/TableView.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+  QuestionCircleOutlined,
+  YoutubeOutlined,
+  ChromeOutlined,
+  GithubOutlined,
+  SettingOutlined,
+} from '@ant-design/icons';
+import { Table, Tag, Tooltip } from 'antd';
+import { GithubUserLink } from 'components';
+import { dateSorter, getColumnSearchProps, tagsRenderer } from 'components/Table';
+import { CourseEvent } from 'services/course';
+import moment from 'moment-timezone';
+import { EventTypeColor, EventTypeToName, TaskTypes } from './model';
+
+type Props = {
+  data: CourseEvent[];
+  timeZone: string;
+};
+
+export function TableView({ data, timeZone }: Props) {
+  return (
+    <Table
+      rowKey={record => (record.event.type === TaskTypes.deadline ? `${record.id}d` : record.id).toString()}
+      pagination={false}
+      dataSource={data}
+      // onRow={() => ({
+      //   onClick: ({ target }) => {
+      //     if (!target.closest('a')) {
+      //       // eslint-disable-next-line no-console
+      //       console.log('row is clicked');
+      //     }
+      //   },
+      // })}
+      columns={[
+        {
+          title: columnFilter(),
+          width: 20,
+          dataIndex: '#',
+          render: (_text, _record, index) => index + 1,
+        },
+        {
+          title: 'Date',
+          width: 120,
+          dataIndex: 'dateTime',
+          render: dateRenderer(timeZone),
+          sorter: dateSorter('dateTime'),
+          defaultSortOrder: 'descend',
+        },
+        { title: 'Time', width: 60, dataIndex: 'dateTime', render: timeRenderer(timeZone) },
+        {
+          title: 'Type',
+          width: 100,
+          dataIndex: ['event', 'type'],
+          render: (value: keyof typeof EventTypeColor) => (
+            <Tag color={EventTypeColor[value]}>{EventTypeToName[value] || value}</Tag>
```

Comment:
Let's use `renderTag` method here. It's declared in `client/src/components/Table/renderers.tsx`

Reply:
renderTag doesn't fit format, create new one in client/src/components/Table/renderers.tsx - eventColorTagRenderer

Reply:
Why can't we do `renderTag(EventTypeColor[value], EventTypeToName[value] || value)`?

Reply:
fix